src/Appwrite/Network/Validator/DNS.php

```diff
@@ -14,34 +15,86 @@ class DNS extends Validator
     public const RECORD_CNAME = 'CNAME';
     public const RECORD_CAA = 'CAA'; // You can provide domain only (as $target) for CAA validation
 
+    protected const FAILURE_REASON_QUERY = 'DNS query failed.';
+    protected const FAILURE_REASON_INTERNAL = 'Internal error occurred.';
+    protected const FAILURE_REASON_UNKNOWN = '';
+
     /**
      * @var mixed
      */
-    protected mixed $logs;
+    protected mixed $logs = [];
 
     /**
-     * @var string
+     * @var array<string>
      */
-    protected string $dnsServer;
+    protected array $dnsServers = [];
+
+    public string $domain = '';
+    public string $resolver = '';
+    public array $recordValues = [];
+    public int $count = 0;
+    public string $reason = '';
 
     /**
      * @param string $target
      */
-    public function __construct(protected string $target, protected string $type = self::RECORD_CNAME, string $dnsServer = '')
+    public function __construct(protected string $target, protected string $type = self::RECORD_CNAME, string $dnsServers = '')
```

Comment:
This validator needs to be in our DNS library and be unit tested. We shouldn't couple validators with Swoole specific features (WaitGroup), we can do that from inside the app context (route/worker/task/etc), would be great to abstract that as well, but this can be a separate problem.

Reply:
for now, let's move it to library, and dont support concurrency there.
then extend it in Appwrite, adding concurrency support for multiple dns servers instead of just one